cdap-data-fabric/src/main/java/co/cask/cdap/data2/sql/SqlTransactionRunner.java

```diff
@@ -60,7 +60,9 @@ public void run(TxRunnable runnable) throws TransactionException {
         connection.rollback();
         throw new TransactionException("Failed to execute the sql queries.", e);
       } catch (SQLException sql) {
-        throw new TransactionException("Failed to rollback a transaction.", sql);
+        TransactionException ex = new TransactionException("Failed to rollback a transaction.", sql);
+        ex.addSuppressed(e);
```

Comment:
seems like it should be the other way round: e.addSuppressed(ex). Because e is the original exception, and ex is or may be misleading as the cause of error.

Reply:
+1